### DISCUSSION TITLE
interaction python ojs

### DISCUSSION THREAD

**Discussion by costadamorte** [quarto-dev/quarto-cli#3387] [Q&A]
Hi. How can I insert python code inside an observable variable? 

{objs}
from numpy import *
v=random.choice([10,20,30,40,50])
r=random.choice([2,3,4,5])  
i ="{0:.2f}".format(v/r)

I= ${i} mA

**Comment by cscheid** [comment]
That's not possible at the moment. We do have support for interaction between Shiny (in R) and Observable, but currently you can't run Python code interactively. You can right Python code at render that and call `ojs_define`, but that will only give you static values:

````
---
title: python+ojs
---

```{python}
import random
v=random.choice([10,20,30,40,50])
r=random.choice([2,3,4,5])
i ="{0:.2f}".format(v/r)
ojs_define(i=i)
```

```{ojs}
I = `${i} mA`
I
```
````

**Reply by costadamorte** [reply]
Ok. Thanks very much
